.github/workflows/reusable-pip-compile.yml

```
@@ -0,0 +1,119 @@
+---
+name: "Refresh pinned dependencies"
+
+"on":
+  workflow_call:
+    # GHA does not support anchors :(
+    # inputs: &inputs
+    inputs:
+      # Commit messae and PR title
+      message:
+        type: string
+        required: true
+      # Branch to create PR from
+      pr-branch:
+        type: string
+        required: true
+      # Branch to base PR on
+      base-branch:
+        type: string
+        required: true
+      # Nox session to call
+      nox-args:
+        type: string
+        required: true
+      # Files to commit
+      changed-files:
+        type: string
+        required: true
+      # Reset branch
+      reset-branch:
+        type: boolean
+        default: false
+  workflow_dispatch:
+    # inputs: *inputs
+    inputs:
+      message:
+        type: string
+        required: true
+      pr-branch:
+        type: string
+        required: true
+      base-branch:
+        type: string
+        required: true
+      nox-args:
+        type: string
+        required: true
+      changed-files:
+        type: string
+        required: true
+      reset-branch:
+        type: boolean
+        default: false
+
+permissions:
+  pull-requests: write
+  contents: write
```

Maybe we could move this to the job level to be stricter?
```suggestion
permissions: {}
```

---

Why? If the parent job doesn't set `permissions`, GHA will fail. It doesn't automatically inherit.

---

It's more of a defensive style. Also, that “parent” only exists through the reusable path, not `workflow_dispatch`.